VimeoNetworking/Sources/AccountStore.swift

```diff
@@ -125,21 +128,21 @@ final class AccountStore
             else
             {
                 let description = "Received corrupted VIMAccount data from keychain"
-                let error = NSError(domain: self.dynamicType.ErrorDomain, code: LocalErrorCode.AccountCorrupted.rawValue, userInfo: [NSLocalizedDescriptionKey: description])
+                let error = NSError(domain: Constants.ErrorDomain, code: LocalErrorCode.accountCorrupted.rawValue, userInfo: [NSLocalizedDescriptionKey: description])
                 
                 throw error
             }
             
-            if let userJSON = account.userJSON as? VimeoClient.ResponseDictionary
+            if let userJSON = account.userJSON
             {
-                try account.user = VIMObjectMapper.mapObject(userJSON) as VIMUser
+                try account.user = VIMObjectMapper.mapObject(responseDictionary: userJSON) as VIMUser
             }
             
             return account
         }
         catch let error
         {
-            _ = try? self.removeAccount(type)
+            _ = try? self.removeAccount(type: type)
```

Comment:
I'm not super sure about the `ofType` changes. Here, for example, you end up with some weirdness:

```Swift
func removeAccount(ofType: AccountType) throws
{
    try self.keychainStore.deleteData(for: ofType.keychainKey())
}
```